src/cppcache/src/ThinClientPoolDM.cpp

```diff
@@ -609,7 +609,8 @@ GfErrType ThinClientPoolDM::sendRequestToAllServers(
   HostAsm::atomicAdd(m_clientOps, 1);
   getStats().setCurClientOps(m_clientOps);
 
-  ACE_Recursive_Thread_Mutex resultCollectorLock;
+  std::shared_ptr<ACE_Recursive_Thread_Mutex> resultCollectorLock(
+      new ACE_Recursive_Thread_Mutex());
```

Comment:
The proper way to allocate a new object into a shared pointer:
```
auto resultCollectorLock = std::make_shared<ACE_Recursive_Thread_Mutex>();
```

This results in a single allocations (combined control block and object) rather than two (control block and object).